src/main.ts

```diff
@@ -24,10 +27,37 @@ export async function run() {
       await cake.bootstrapScript(scriptPath, toolsDir);
     }
 
-    await cake.runScript(scriptPath, toolsDir, target, verbosity);
+    await cake.runScript(
+      scriptPath,
+      toolsDir,
+      target,
+      verbosity,
+      ...scriptArgs
+    );
   } catch (error) {
     core.setFailed(error.message);
   }
 }
 
+function getScriptArguments(): CakeArgument[] {
+  const result: CakeArgument[] = [];
+  const input: string = core.getInput("script-arguments");
+  if (!input) {
+    return [];
+  }
+
+  const options = yargsParser(input);
+  const optionsKeys: string[] = Object.keys(options);
+  optionsKeys.forEach((optionsKey) => {
+    // we ignore that because of yargs-parser implementation
+    if (optionsKey === "_") {
+      return;
+    }
+
+    const cakeArgument = new CakeArgument(optionsKey, options[optionsKey]);
```

Comment:
I have added handling of CakeSwitch. 
In different way, 
I decided to check if the parameter is true, because there is no possibility to pass false into CakeSwitch.